pandas/core/indexing.py

```diff
@@ -1266,9 +1278,23 @@ def _convert_to_indexer(
                 inds, = obj.nonzero()
                 return inds
             else:
-                # When setting, missing keys are not allowed, even with .loc:
-                kwargs = {"raise_missing": True if is_setter else raise_missing}
-                return self._get_listlike_indexer(obj, axis, **kwargs)[1]
+                try:
+                    # When setting, missing keys are not allowed, even with # .loc:
+                    kwargs = {"raise_missing": True if is_setter else raise_missing}
```

Comment:
you don't want to do this in a try/except as its extremely error prone. Instead if you have a is_setter you can just make sure the columns exist (but doing the _convert_to_indexer).

I think if you do this *first* e.g. almost first in _setitem_with_indexer you won't need to make the code change you have above about taking the split path.

Reply:
I force `take_split_path = True` to work around #27583. Otherwise I would fail tests such as `test_setitem_list_all_missing_columns_scalar` in  `pandas/tests/frame/test_indexing.py`.